### PR TITLE
CLOUD-220 One-pod test

### PR DESCRIPTION
**Pull request by cap1984**
One pod test with backup and restore procedure

### CODE REVIEW

File: e2e-tests/one-pod/conf/on-demand-backup-s3.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: psmdb.percona.com/v1
```
Comment (delgod) [inline e2e-tests/one-pod/conf/on-demand-backup-s3.yml]: please use name and content identical to `e2e-tests/demand-backup/conf/backup-minio.yml`

File: e2e-tests/one-pod/conf/one-pod.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      persistentVolumeClaim:
+        resources:
+          requests:
+            storage: 3Gi
+  mongod:
+    net:
+      port: 27017
+      hostPort: 0
+    security:
+      redactClientLogData: false
```
Comment (delgod) [inline e2e-tests/one-pod/conf/one-pod.yml]:
```suggestion
      redactClientLogData: false
      enableEncryption: false
```

File: e2e-tests/one-pod/run [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+spinup_psmdb() {
+    local cluster=$1
+    local config=$2
+    local size="${3:-3}"
+
+    desc 'create first PSMDB cluster'
+    kubectl apply \
+        -f $conf_dir/secrets.yml \
+        -f $conf_dir/client.yml \
+        -f $config
```
Comment (delgod) [inline e2e-tests/one-pod/run]:
```suggestion
    apply_cluster $config
```

File: e2e-tests/one-pod/conf/s3-secrets.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
```
Comment (delgod) [inline e2e-tests/one-pod/conf/s3-secrets.yml]: please use `e2e-tests/conf/minio-secret.yml` file

File: e2e-tests/one-pod/run [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    desc 'check data after backup'
+    compare_mongo_cmd "find" "myApp:myPass@one-pod-rs0.${namespace}"
+}
+
+main() {
+    create_namespace $namespace
+    deploy_operator
+    kubectl apply \
+        -f $conf_dir/cloud-secret.yml
```
Comment (delgod) [inline e2e-tests/one-pod/run]: AWS and GCP secrets not needed if we use minio

File: e2e-tests/one-pod/run [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+    cluster="one-pod"
```
Comment (delgod) [inline e2e-tests/one-pod/run]:
```suggestion
    cluster="one-pod-rs0"
```

File: e2e-tests/one-pod/run [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        -f $config
+
+    desc 'check if all 3 Pods started'
+    wait_for_running "$cluster-rs0" "$size"
+    sleep 20
+
+    desc 'write data'
+
+    run_mongo 'db.createUser({user: "myApp", pwd: "myPass", roles: [{ db: "myApp", role: "readWrite" }]})' \
+              "userAdmin:userAdmin123456@one-pod-rs0.${namespace}"
```
Comment (delgod) [inline e2e-tests/one-pod/run]: please remove hardcoded cluster name everywhere
```suggestion
              "userAdmin:userAdmin123456@${cluster}.${namespace}"
```

File: e2e-tests/functions [hunk 8]
```diff
@@ -204,6 +208,29 @@ deploy_helm() {
     fi
 }
 
+deploy_local_s3() {
```
Comment (delgod) [inline e2e-tests/functions]:
```suggestion
deploy_minio() {
```

File: e2e-tests/one-pod/run [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    wait_restore_object ${backup1}
+
+    compare_mongo_cmd "find" "myApp:myPass@one-pod-rs0.${namespace}"    
+
+    kubectl logs pod/one-pod-backup-coordinator-0
+    kubectl delete -f "$test_dir/conf/${backup1}.yaml"
+    wait_for_running "$cluster-rs0" 1
+
+    desc 'check data after backup'
+    compare_mongo_cmd "find" "myApp:myPass@one-pod-rs0.${namespace}"
```
Comment (delgod) [inline e2e-tests/one-pod/run]: why we need the same command in 57 and 64 lines?

File: e2e-tests/one-pod/conf/one-pod.yml (hunk 2 above)
Comment (cap1984) [reply e2e-tests/one-pod/conf/one-pod.yml]: done

File: e2e-tests/one-pod/run (hunk 3 above)
Comment (cap1984) [reply e2e-tests/one-pod/run]: done

File: e2e-tests/one-pod/run (hunk 5 above)
Comment (cap1984) [reply e2e-tests/one-pod/run]: done

File: e2e-tests/one-pod/run (hunk 6 above)
Comment (cap1984) [reply e2e-tests/one-pod/run]: done

File: e2e-tests/one-pod/conf/s3-secrets.yml (hunk 4 above)
Comment (cap1984) [reply e2e-tests/one-pod/conf/s3-secrets.yml]: done

File: e2e-tests/one-pod/run (hunk 7 above)
Comment (cap1984) [reply e2e-tests/one-pod/run]: done

File: e2e-tests/one-pod/run (hunk 9 above)
Comment (cap1984) [reply e2e-tests/one-pod/run]: removed the one in the middle

File: e2e-tests/functions (hunk 8 above)
Comment (cap1984) [reply e2e-tests/functions]: done

File: e2e-tests/one-pod/conf/on-demand-backup-s3.yml (hunk 1 above)
Comment (cap1984) [reply e2e-tests/one-pod/conf/on-demand-backup-s3.yml]: done

File: e2e-tests/one-pod/run [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+    desc 'check if Pod is started'
+    wait_for_running "${cluster}" "$size"
+    sleep 20
+
+    desc 'write data'
+
+    run_mongo 'db.createUser({user: "myApp", pwd: "myPass", roles: [{ db: "myApp", role: "readWrite" }]})' \
+              "userAdmin:userAdmin123456@${cluster}.${namespace}"
+}
```
Comment (delgod) [inline e2e-tests/one-pod/run]: mandatory to add data into the collection before backup
```suggestion

    run_mongo 'use myApp\n db.test.insert({ x: 100500 })' \
        "myApp:myPass@${cluster}.${namespace}"
}
```

File: e2e-tests/one-pod/run [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    local cluster=$1
+    local config=$2
+    local size="${3:-3}"
+
+    desc 'create first PSMDB cluster'
+    apply_cluster $config
+
+    desc 'check if Pod is started'
+    wait_for_running "${cluster}" "$size"
+    sleep 20
```
Comment (delgod) [inline e2e-tests/one-pod/run]: I think sleep not needed here
but I want to control that TLS and encryption are disabled, so it is needed to compare statefulset content
```suggestion
    compare_kubectl statefulset/$cluster
```

Comment (delgod) [reply e2e-tests/one-pod/run]: is sleep really needed?

File: e2e-tests/run [hunk 12]
```diff
@@ -9,6 +9,7 @@ $dir/init-deploy/run
 $dir/limits/run
 $dir/scaling/run
 $dir/monitoring/run
+$dir/monitoring/one-pod/run
```
Comment (delgod) [inline e2e-tests/run]:
```suggestion
$dir/one-pod/run
```

File: e2e-tests/one-pod/run (hunk 11 above)
Comment (cap1984) [reply e2e-tests/one-pod/run]: it takes time to make new mongo user available inside MongoDB. If we remove sleep we'll face auth error.

**Pull request merged**